cmds/scp/scp.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2012-2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Scp copies files between hosts on a network.
+//
+// Synopsis:
+//     scp [-t|-f] [FILE]
+//
+// Description:
+//     If -t is given, decode SCP protocol from stdin and write to FILE.
+//     If -f is given, stream FILE over SCP protocol to stdout.
+//
+// Options:
+//     -t: Act as the target
+//     -f: Act as the source
+//     -v: Passed if SCP is verbose, ignored
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+)
+
+const (
+	SUCCESS = 0
+)
+
+var (
+	isTarget = flag.Bool("t", false, "Act as the target")
```

Comment:
is this the standard? We have two flags, four combinations, which is a bit weird, since we only need two? If this is their idea, it's kind of weird :-)

Reply:
Sadly it's "standard". The way this works is that when you run "scp my-uroot-system:some-file ." scp on your machine will become a sink on stdin, and scp will will ssh into my-uroot-system and run "scp -f some-file". If it's instead is "scp some-file my-uroot-system:" the command will be "scp -t .". Writing this I realized I forgot to handle the path and it will always write to the cwd, oops!

Anyway, this means that we need to adhere to a "standard" in terms of how our command line is parsed.

It might not be clear, but my intention was only to implement support for receiving scp commands - you will for example not be able to ssh to the uroot system and initiate an scp from inside the uroot system with this patch. It's not that hard to add on, it's just more work.